File: src/main/java/com/domain/DbUser.java
package com.domain;

import javax.persistence.*;
import java.sql.Timestamp;

/**
 * 用户类
 *
 * @author Five
 * @date create in 下午2:30 2017/11/30
 */

@Entity
@Table(name = "user")
public class DbUser {
    @Id
    @GeneratedValue
    private Integer id;

    @Column(length = 32)
    private String uid;

    @Column(length = 64, nullable = true)
    private String realname;

    @Column(length = 64, nullable = true)
    private String nickname;

    private Integer gender;

    @Column(nullable = true)
    private Timestamp birthday;

    @Column(length = 11, nullable = true)
    private String phone;

    @Column(length = 64, nullable = true)
    private String email;

    private Integer status;

    @Column(updatable = false)
    private Timestamp createtime;

    /**
     * 构造函数
     */
    public DbUser() {
    }
}

File: src/main/java/com/domain/entity/DbUser.kt
package com.domain.entity

import java.util.*
import javax.persistence.*

/**
 * class
 *
 * @author Five0314
 * @date create in 下午2:29 2017/12/14
 */
@Entity
@Table(name = "user")
class DbUser {
    @Id
    @GeneratedValue
    var id: Int = 0 // 数据库ID

    @Column(length = 32)
    var uid: String = "" // 用户编号

    @Column(length = 64)
    var nickname: String? = null // 昵称

    var gender: Int = 0 // 性别

    @Column(nullable = true)
    var birthday: Date? = null // 出生日期

    @Column(length = 11)
    var phone: String? = null // 电话

    @Column(length = 64)
    var email: String? = null // 邮箱

    var status: Int = 0 // 状态

    @Column(updatable = false)
    lateinit var createtime: Date // 创建时间
}